### PR TITLE
feat(skore): Create a "custom" metrics registry under each report

### PR DESCRIPTION
**Pull request by thomass-dev**
> [!CAUTION]
> Just a notebook for reflection for now.

Points to cover:
- redefine `__call__` in each metric subclass to provide a more user-friendly API
    - equivalent of the current implementation with `r2` calling `_r2`
- show the docstr from the metric `r2 = R2(); help(r2)`
- show the helper
    - standard metrics
    - custom metrics
- deal with CrossValidation (only metrics)

---

Closes https://github.com/probabl-ai/skore-hub/issues/940.

### CODE REVIEW

**Comment by thomass-dev** [comment]
```python
from sklearn.datasets import make_regression
from sklearn.linear_model import LinearRegression
from sklearn.metrics import get_scorer
from sklearn.model_selection import train_test_split

from skore._sklearn._metric.registry import NewEstimatorReport

X, y = make_regression(random_state=42)
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42)

report = NewEstimatorReport(
    LinearRegression(),
    X_train=X_train,
    y_train=y_train,
    X_test=X_test,
    y_test=y_test,
)

def my_business_metric(*args, **kwargs):
    return 1

report.metrics.append(get_scorer("explained_variance"))
report.metrics.append("neg_mean_squared_error")
report.metrics.append(my_business_metric)

print(report.metrics)
print(report.metrics.r2())
print(report.metrics.explained_variance_score())
print(report.metrics.mean_squared_error())
print(report.metrics.my_business_metric())
```

**Comment by auguste-probabl** [comment]
Superseded by #2736
